lib/driver.js

```diff
@@ -515,6 +525,74 @@ class XCUITestDriver extends BaseDriver {
     // finally, return true since the superclass check passed, as did this
     return true;
   }
+
+  async installApp () {
+    if (this.isSafari()){
+      return;
+    }
+    // if user has passed in desiredCaps.autoLaunch = false
+    // meaning they will manage app install / launching
+    if (this.opts.autoLaunch === false) {
+      return;
+    }
+
+    if (this.isRealDevice()) {
+      await this.installToRealDevice();
+    } else {
+      await this.installToSimulator();
+    }
+  }
+
+  async installToSimulator () {
+    if (this.opts.fullReset && this.opts.bundleId) {
+      await removeApp(this.opts.device.udid, this.opts.bundleId);
+    }  
+    await installApp(this.opts.device.udid, this.opts.app);
+  }
+
+  async installToRealDevice () {
+    if (this.opts.udid) {
+      if (await this.opts.device.isInstalled(this.opts.bundleId)) {
+        log.debug("App is installed.");
+        if (this.opts.fullReset) {
+          log.debug("fullReset requested. Forcing app install after uninstall app");
+          await this.opts.device.remove(this.opts.bundleId);
+        } else {
+          log.debug("fullReset not requested. No need to install.");
+          return;
+        }
+      } else {
+        log.debug("App is not installed. Will try to install.");
+      }
+
+      if (this.opts.app) {
+        await this.opts.device.install(this.opts.app);
+        log.debug('App installed.');
+      } else {
+        log.debug("Real device specified but no ipa or app path, assuming bundle ID is " +
+                     "on device");
+      }
+    } else {
+      log.debug("No device id or app, not installing to real device.");
+    }
+  }
+
+  async getIDeviceObj () {
+    log.debug(`Creating IOSDeploy object with udid ${this.opts.udid}`);
+    try {
+      //keeping this iDevice for future once we have ideviceinstaller working for ios 10
+      iDevice(this.opts.udid);
```

Comment:
I think it would be better to remove it, and just leave a comment about it being there later.

Another option is to provide a generic interface that could be the `iDevice` object or the `IOSDeploy` stuff.